torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -768,14 +768,11 @@ def pad_bounding_box(
 
     left, right, top, bottom = _parse_pad_padding(padding)
 
-    bounding_box = bounding_box.clone()
-
-    # this works without conversion since padding only affects xy coordinates
-    bounding_box[..., 0] += left
-    bounding_box[..., 1] += top
     if format == features.BoundingBoxFormat.XYXY:
-        bounding_box[..., 2] += left
-        bounding_box[..., 3] += top
+        pad = [left, top, left, top]
+    else:
+        pad = [left, top, 0, 0]
+    bounding_box = bounding_box + torch.tensor(pad, dtype=bounding_box.dtype, device=bounding_box.device)
```

Comment:
Why is this (and the substraction below) out-of-place rather than in-place?

Reply:
We should avoid modifying input inplace. Previously, we were cloning and doing inplace but it is slower. See https://github.com/pytorch/vision/pull/6881#discussion_r1012228887

Reply:
If the intention is to not modify in place the data, it's faster to do out of place than clone and do inplace.

Reply:
@datumbox I was quicker )))

Reply:
fair play :D

Reply:
Also, yes, out of place ops should always be faster than clone + in-place, as in the latter ones you read the data twice.